internal/config/image/cuda_image.go

```diff
@@ -78,9 +89,14 @@ func (i CUDA) HasEnvvar(key string) bool {
 // image is considered legacy if it has a CUDA_VERSION environment variable defined
 // and no NVIDIA_REQUIRE_CUDA environment variable defined.
 func (i CUDA) IsLegacy() bool {
-	legacyCudaVersion := i.env[EnvVarCudaVersion]
-	cudaRequire := i.env[EnvVarNvidiaRequireCuda]
-	return len(legacyCudaVersion) > 0 && len(cudaRequire) == 0
+	cudaVersion := i.Getenv(EnvVarCudaVersion)
```

Comment:
What was the reason for the switch to `Getenv`? I understand that we may be using things inconsistently here, but let's rather address that as a follow-up to keep the refactoring diff smaller?